### PR TITLE
FEAT:- create request validation schemas using JOI for every request, also validate the incoming request before moving ahead.

### PR DESCRIPTION
**Pull request by BhawnaJain01**
Added validation using joi .

### CODE REVIEW

File: server/src/controllers/github.controller.ts [hunk 1]
```diff
@@ -1,19 +1,31 @@
 import { Request, Response } from "express";
 import { v4 } from "uuid";
 import { insert_workflow } from "../services/workflow";
+import Joi from "joi";
```
Comment (sahilarora02) [inline server/src/controllers/github.controller.ts:4]: why there is no package.json file with 'Joi' as dependency??

File: server/src/controllers/github.controller.ts [hunk 2]
```diff
@@ -1,19 +1,31 @@
...
+     });
+  }
   const { repo, owner, access_token, repo_url, user_id, additional_email } =
     req.body;
   const webhook_secret = process.env.WEBHOOK_SECRET;
   const baseUrl = process.env.BASE_URL;
 
   try {
+
+    console.log("i am in try");
```
Comment (sahilarora02) [inline server/src/controllers/github.controller.ts:28]: Use Logger 

also use them only while development

File: server/src/routes/workflow.router.ts [hunk 3]
```diff
@@ -3,6 +3,6 @@ import { get_workflows} from "../controllers";
 
 const workflowRouter = express.Router();
 
-workflowRouter.get("/:user_id", get_workflows);
+workflowRouter.get("/:user_id?", get_workflows);
```
Comment (sahilarora02) [inline server/src/routes/workflow.router.ts:6]: why making it optional???

Comment (BhawnaJain01) [reply server/src/routes/workflow.router.ts:6]: removed the user_id validation.

**Review by sahilrohera10 — APPROVED** [review]

**Pull request merged**